src/core/utils/channel_manager.cpp

```diff
@@ -137,6 +162,25 @@ void ChannelManager::PreparePendingDataset(void)
         ExitNow();
     }
 
+    // `pendingActiveTimestamp` will be non-zero if the Pending
+    // Dataset is valid and is performing the same channel change.
+    // We check to ensure its timestamp is indeed ahead of current
+    // Active Dataset's timestamp, and if so, we skip updating
```

Comment:
From the code below, it seems we skip this time but set a timer and check again later.

Reply:
The `mState` is also changed to `kStateSentMgmtPendingDataset`. The timer is then used as a timeout to verify/check that we did go to new channel (if we don't within the timeout then we restart the whole process again).  We are basically treating the current Pending Dataset as the one we wanted to prepare and submit.